utils/useDebounceEffect.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect, DependencyList } from "react"
+
+export function useDebounceEffect(
```

Comment:
Maybe a comment on what this function is used for would be helpful! Utils are very specific to a functionality and context helps understand what this is for.